src/Compilers/Core/Portable/PEWriter/DebugSourceDocumentLocationComparer.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.Cci
+{
+    internal sealed class DebugSourceDocumentLocationComparer : IComparer<DebugSourceDocument>
+    {
+        public static DebugSourceDocumentLocationComparer Instance
+            => new DebugSourceDocumentLocationComparer();
+
+        private DebugSourceDocumentLocationComparer()
+        {
+        }
+
+        int IComparer<DebugSourceDocument>.Compare(DebugSourceDocument x, DebugSourceDocument y)
+            => x.Location.CompareTo(y.Location);
```

Comment:
Fixed - replaced with string.Compare overload that performs case-insensitive and culture-agnostic comparison.